### ISSUE TITLE
Sample python gui shadow can cost lots of AWS IoT $ charges if you leave it running

### ISSUE THREAD

**Issue by ghost** [MicrochipTech/aws-iot-zero-touch-secure-provisioning-kit#2]
The line of code:

 `self.after(500,` self.on_update)

in the file "aws_interact_gui.py" causes 2 messages per second to the AWS IoT network. 

I highly recommend that users change the 500ms to something higher and then check their usage.
Unfortunately raising the timer will also delay the update of the gui button status display as this all one polling loop.

One good way to check usage is to turn on AWS IoT Cloudwatch.

**Comment by DanielHarris-Microchip** [comment]
Hi msymmes,

Good catch!  You're right, if one does leave the "aws_interact_gui.py" script running, it can wrack up an inordinate number of IoT messages to AWS.  As is often the case, things need to be delivered in a short amount of time, and the script was written hastily.  A better way to do it would be to have the script subscribe to the MQTT update topic for the thing - that way updates are pushed from AWS instead of polled.  I plan on implementing and pushing the update early this this week.

Cheers,
Daniel

**Comment by msymmes** [comment]
Daniel,

Thanks for your reply.   I am trying to use the OLED pixel display today
but can't get it to display any text.

In main.c, I added 2 lines but with no outcome whatsoever.

I added after:   oled1_init();

ssd1306_init();
ssd1306_write_text("Hello");

This is somewhat important to me because of the demo I want to do for a big
company.

BTW, I have IoT and Lambda logging and passing payloads thru AWS SNS. Press
a button and receive an email.


Mark Symmes
AWS Certified  Solutions Architect



Cloud9Architects.io
Toronto, Ontario
416-558-6275
